limits/filetransfer/limits.go

```diff
@@ -373,8 +377,7 @@ func updateCounters() {
 	tick := time.NewTicker(time.Duration(countersUpdateSleepTimeS) * time.Second)
 	defer tick.Stop()
 	for counterUpdateRunning {
-		select {
-		case <-tick.C:
+		for range tick.C {
```

Comment:
was this also required by the static checker? as far as I know it should work, its just this part of code was tested manually for such a long time, including some graphs of the tx/rx from debug info :) you know. if required let's have it.

Reply:
I was unsure about this one too. The lining error is:
```
limits/filetransfer/limits.go:380:3: S1000: should use a simple channel send/receive instead of `select` with a single case (gosimple)
		select {
		^
```

A safer patch that also lints can be:
```
diff --git a/limits/filetransfer/limits.go b/limits/filetransfer/limits.go
index 204ec99..e580f45 100644
--- a/limits/filetransfer/limits.go
+++ b/limits/filetransfer/limits.go
@@ -377,22 +377,21 @@ func updateCounters() {
        tick := time.NewTicker(time.Duration(countersUpdateSleepTimeS) * time.Second)
        defer tick.Stop()
        for counterUpdateRunning {
-               for range tick.C {
-                       txTot := deviceCounters.bytesTransferred
-                       rxTot := deviceCounters.bytesReceived
-                       rx := rxTot - lastRX
-                       tx := txTot - lastTX
-                       lastRX = rxTot
-                       lastTX = txTot
-                       // avg[n+1] = w * Y[n] + (1 - w) avg[n]
-                       avgRX = w1*float64(rx) + _w1*avgRX
-                       avgTX = w1*float64(tx) + _w1*avgTX
-
-                       countersMutex.Lock()
-                       deviceCounters.bytesTransferredAvg1m = avgTX
-                       deviceCounters.bytesReceivedAvg1m = avgRX
-                       countersMutex.Unlock()
-               }
+               <-tick.C
+               txTot := deviceCounters.bytesTransferred
+               rxTot := deviceCounters.bytesReceived
+               rx := rxTot - lastRX
+               tx := txTot - lastTX
+               lastRX = rxTot
+               lastTX = txTot
+               // avg[n+1] = w * Y[n] + (1 - w) avg[n]
+               avgRX = w1*float64(rx) + _w1*avgRX
+               avgTX = w1*float64(tx) + _w1*avgTX
+
+               countersMutex.Lock()
+               deviceCounters.bytesTransferredAvg1m = avgTX
+               deviceCounters.bytesReceivedAvg1m = avgRX
+               countersMutex.Unlock()
        }
 }
 
```

What do you think about this second one? That is basically removing a single case select but blocking the same way it was (which is what I am trying to do with the `for range` too

Reply:
makes sense!